---
layout: post  
title: Protocol for 16S Amplicfication of coral and reef sediment with 515F and 806R V4 primers  
date: '2022-06-21'  
categories: Notebook  
tags: Hawaii 2022, Bhattacharya Lab, PCR, Microbiome biogeography, Montipora capitata  
projects: Bhattacharya Lab  
---


--- 
***The goal of running this procedure is to capture the V4 16S rDNA region of the microbial community in reef sediment for amplicon sequencing.***

---

## Materials  and Methods
- Q5® Hot Start High-Fidelity 2X Master Mix ([link](https://www.neb.com/products/m0494-q5-hot-start-high-fidelity-2x-master-mix#Product%20Information))  
- Invitrogen™ UltraPure™ DNase/RNase-Free Distilled Water ([link](https://www.fishersci.com/shop/products/invitrogen-ultrapure-dnase-rnase-free-distilled-water-1/10977023?ef_id=Cj0KCQjw2MWVBhCQARIsAIjbwoOfJUlLEvFyQ3UwofU1N_2spcOoN0fwEWrWqya9uM5h3DLkboodycwaAn4rEALw_wcB:G:s&ppc_id=PLA_goog_16713066638__10977023___10153541525911855182&ev_chn=shop&s_kwcid=AL!4428!3!!!!x!!10977023&gclid=Cj0KCQjw2MWVBhCQARIsAIjbwoOfJUlLEvFyQ3UwofU1N_2spcOoN0fwEWrWqya9uM5h3DLkboodycwaAn4rEALw_wcB))  
- Forward and Reverse Amplification Primers ([link](https://earthmicrobiome.org/protocols-and-standards/16s/))  
	- FWD: 515F (Parada et al., 2016) GTGYCAGCMGCCGCGGTAA  
	- REV: 809R (Apprill et al., 2015) GGACTACNVGGGTWTCTAAT  
- Eppendof Mastercycler (catalog number 5333)  
- Purified template DNA  
- Pipetters and tips (p1000, p200, p10), multichannel is helpful but not necessary  
- 0.2mL PCR Tube Strips

### Samples

Take note of the following parameters in your lab notebook. Dilute all samples to 1 ng/µL, with a total quantity of 20µL. This [PhysWeb calculator](https://www.physiologyweb.com/calculators/dilution_calculator_mass_per_volume.html) is a helpful tool for calculating dilutions. In many cases, the DNA will already be extracted for the samples. This [spreadsheet](https://docs.google.com/spreadsheets/d/1jKOYeW-VEpQAaw0l1E0fzl9vYXP9IfaG/edit?usp=sharing&ouid=106701088267677384349&rtpof=true&sd=true) has a list of all HI22 microbiome samples and their DNA conc. immediately after extraction. 

Tests were run on:

 | Extraction Date  |  Sample ID | Type | DNA Quantity (ng/µL) | Stock (µL) | Buffer (ultrapure H20; µL) |  
 | --- | --- | --- | --- | --- | --- | 
| 20220618 | S5 | Reef Sediment | 1.57 | 12.74 | 7.26 |  
| 20220618 | S23 | Reef Sediment | 11.55 | 1.73 | 18.27 |  
| 20220618 | S11 | Reef Sediment | 6.09 | 3.29 | 16.71 |  
| 20220618 | C61 | Mcap tissue | 49.2 | 0.41 | 19.59 |  
| 20220618 | C76 | Mcap tissue | 18.9 | 1.06 | 18.94 |  
| 20220618 | C33 | Mcap tissue  | 51.0 | 0.39 | 19.61 |  
| 20220618 | G93 | Gfas tissue | 25.0 | 0.80 | 19.20 |  
| 20220618 | G94 | Gfas tissue | 23.7 | 0.84 | 19.16 |  
| 20220618 | G95 | Gfas tissue | 17.0 | 1.18 | 18.82 |  
| 20220618 | Blank | Ultrapure Water | 0 | N/A | N/A | 

### PCR Reaction Mix  
To create enough solution for **three aliquots** of 7 samples, a blank, plus a little extra, I used the following recipe:  
- 390 µL Q5 MasterMix
- 39 µL 10µM 515F Forward Primer
- 39 µL 10µM  806R Forward Primer  
- 156 µL ultrapure water

The solution above was aliquoted into 0.2mL PCR Tube Strips (20µL each), then 5µL of template DNA was added to each tube.

### Thermocycling parameters  
| Step | Temp | Time |  
| --- | --- | --- |  
| Initial denaturation | 98°C | 00:00:30 |  
| Denaturation | 98°C | 00:00:15 |  
| Annealing | 55°C | 00:01:00 |  
| Extension | 72°C | 00:00:30 |  
| Cycle | GOTO 2 | REP 35 |  
| Final extension | 72°C | 00:10:00 |  
| Hold | HOLD: 4°C | not specified |

### Purification

## Results


### Gel  
- Followed Bhattacharya Lab Agarose Gel protocol for a medium 1% agarose gel ([link](https://echille.github.io/E.-Chille-Open-Lab-Notebook/Bhattacharya-Lab-Gel-Electrophoresis-Protocols/))  with the Thermo Scientific GeneRuler 1 kb Plus DNA Ladder 

### Purfied Product Quanity and Purity

## Notebook Entry 20220621**  
![]()
